pypeit/core/gui/gui_util.py

```diff
@@ -0,0 +1,257 @@
+                        else self.observers_descr[action]
+            print(f'    {action}: {descr}')
+        print('-'*50)
+
+
+class UpdateableRangeSlider(widgets.RangeSlider):
+    """
+    A range slider.
+
+    This is virtually identical to the base class, but with a few customizations
+    regarding where labels are placed (or removed).
+    """
+    def __init__(self, ax, label, valmin, valmax, valinit=None, valfmt=None, closedmin=True,
+                 closedmax=True, dragging=True, valstep=None, orientation="horizontal",
+                 track_color='lightgrey', handle_style=None, **kwargs):
+
+        super().__init__(ax, label, valmin, valmax, valinit=valinit, valfmt=valfmt,
+                         closedmin=closedmin, closedmax=closedmax, dragging=dragging,
+                         valstep=valstep, orientation=orientation, track_color=track_color,
+                         handle_style=handle_style, **kwargs)
+        self.label.set_position((0.5, 0.8))
+        self.label.set_verticalalignment('bottom')
+        self.label.set_horizontalalignment('center')
+        self.label.set_weight('bold')
+        # "Removes" the labels showing the current range
+        self.valtext.set_visible(False)
+
+    def update_range(self, rng, label=None):
+        """
+        Update the slider to cover a different range.
+        """
+        self._active_handle = None
+        xy = self.poly.get_xy()
+        xy[:,0] = np.roll(np.repeat(rng, 3)[:-1],-1)
+        self.poly.set_xy(xy)
+        self.valmin, self.valmax = rng
+        self.valinit = np.array(rng)
+        self._handles[0].set_xdata(np.array([rng[0]]))
+        self._handles[1].set_xdata(np.array([rng[1]]))
+        self.ax.set_xlim(rng)
+        if label is not None:
+            self.label.set_text(label)
+        self.set_val(rng)
+
+
+class UpdateableImage:
+    """
+    Provides an interface to change the Z range of an image and toggle between a
+    set of images.
+
+    Args:
+        images (`numpy.ndarray`_, :obj:`list`):
+            One or more images to plot.  The images should have the same shape.
+        image_plot (`matplotlib.image.AxesImage`_):
+            Object returned by `matplotlib.pyplot.imshow`_.
+        slider (:class:`UpdateableRangeSlider`):
+            Slider used to adjust the image plot limits.
+    """
+    def __init__(self, images, image_plot, slider):
+        self.showing = 0
+        self.images = images if isinstance(images, list) else [images]
+        self.nimages = len(self.images)
+        self.image_plot = image_plot
+        self.slider = slider
+        self.slider.on_changed(self.change_range)
+
+    def change_range(self, val):
+        """
+        Adjust the plotted range.
+
+        Args:
+            val (:obj:`list`):
+                New range to plot
+        """
+        self.image_plot.set_clim(*val)
+        pyplot.draw()
+
+    def next_image(self, *args, **kwargs):
```

Comment:
Also here. Why do we need to include the args/kwargs in some places?

Reply:
Same as `update_traces` in `EdgeInspectorGUI`.  I updated the calling sequence (we only need *args), and added the same note to the doc string.